### ISSUE TITLE
Overly complicated rating system

### ISSUE THREAD

**Issue by AeliusSaionji** [hydrusnetwork/hydrus#154]
Two different ideas
- Stat collection. What images does the user skip during slideshows? Which ones do they look at until the slideshow advances itself? Which tags does the user most often search for? Which images has the user manually searched for and opened?
- Relative rating system. Select an image, click "rate this image relative to other images". Hydrus viewer then goes into split-screen. On the left (or topleft corner), the selected image. The rest of the viewer produces random images (perhaps, optionally images with related tags, or visual similarity) and the user rates each randomly produced image as better, worse, or equal to the selected image.

For either of these approaches, this data can be used to sort images.

**Comment by pokeball99** [comment]
1. do you mean stats being sent to hydrus dev, or just local stats? also if its the latter, i say we don't have it put into the database file, so that if people share there databases, they wont be giving that shit out.
2. neat idea, no idea how the dev would think of it

**Comment by Link-Satonaka** [comment]
Yep. Local only- not for hydrus dev, not available outside of client.exe

**Comment by Link-Satonaka** [comment]
Why are you sharing your database, as opposed to exporting?

**Comment by mserajnik** [comment]
Hey, issues here have been reenabled and will be used for bug reports/feature requests in the future.

We are currently clearing out all the existing issues from 2017 and prior.
If your bug report or feature request is still relevant, please create a new issue (or use one of the other channels; see https://hydrusnetwork.github.io/hydrus/).

**Comment by bbappserver** [comment]
@CuddleBear92 
This is still an issue. A rating system from hydrus watching your viewing time would not go amiss.

**Comment by mserajnik** [comment]
@bbappserver Reopened.

**Comment by vickyorlo** [comment]
The second point in this issue is especially hilarious to me, considering I've used hydrus long enough to know that such a "relative rating system" existed - it was called a rating filter and worked very similarly to the current archive/delete filter. Of course, it was incredibly slow - rating 100 files would take about 500 choices, or even more if you used the "they're the same" option. And you would, because comparing two files in terms of subjective quality is either obvious or impossible, with very little inbetween.

Also, view time and some other stats are already implemented.

**Comment by ShadowJonathan** [comment]
This is reminding me of that one scene from the "Social Network", are you saying you want to implement such an algorithm like that that'd balance out ratings compared to "which is better" comparisons?

**Comment by imtbl** [comment]
> I've used hydrus long enough to know that such a "relative rating system" existed - it was called a rating filter and worked very similarly to the current archive/delete filter

That's good to know; then there's probably a reason why HyDev removed it. Will keep it in mind when talking to him about it.

**Comment by vickyorlo** [comment]
If I recall right, it was half because it was borderline useless, half because it was an awful hackjob. The filter subsystem got refactored to support custom actions and shortcuts and there was meant to be more filters, but only the A/D filter remained.

**Comment by DonaldTsang** [comment]
@AeliusSaionji @vickyorlo 
That type of system is completely based on these algorithms
- https://github.com/sublee/trueskill
- https://github.com/sublee/glicko2
- https://github.com/sublee/elo
- Alternatives https://github.com/McLeopold/PythonSkills

And yes, they are originally meant for Chess and Go. But the problem is that
A. you need to do a lot of comparisons to be accurate
B. there are no shortcuts, tag correlations are objective stats while relative ratings are subjective.
C. People change in tastes.

**Comment by DonaldTsang** [comment]
An alternative would be a ranking system based on tag preferences, which is more objective.

**Comment by Artoch47** [comment]
A small project that uses Glicko-2 to rate files - https://github.com/onlaj/Kura
Maybe something like this could be implemented into Hydrus.

Alternative algorithm: https://github.com/pfmonville/whole_history_rating
